ipn/ipnlocal/local.go

```diff
@@ -5537,6 +5537,41 @@ func (b *LocalBackend) CheckUDPGROForwarding() error {
 	return warn
 }
 
+// SetUDPGROForwarding enables UDP GRO forwarding for the default network
+// interface of this machine. It can be done to improve performance for nodes
+// acting as Tailscale subnet routers or exit nodes. Currently (9/5/2024) this
+// functionality is considered experimental and only safe to use via explicit
+// user opt-in for ephemeral devices, such as containers.
+// https://tailscale.com/kb/1320/performance-best-practices#linux-optimizations-for-subnet-routers-and-exit-nodes
+func (b *LocalBackend) SetUDPGROForwarding() error {
+	if b.sys.IsNetstackRouter() {
+		return errors.New("UDP GRO forwarding cannot be enabled in userspace mode")
+	}
+	tunSys, ok := b.sys.Tun.GetOK()
+	if !ok {
+		// TODO: perhaps return a non-error signal that the call needs to be retried?
+		return errors.New("unable to retrieve tun device configuration")
```

Comment:
Do we expect a scenario where this will happen and it should be retried? i.e. is there a race between serving localapi and TUN init? If we need a TODO it should be assigned.

Same for the other TODOs in this func. If there's no race between initializing the components and serving localapi I don't see a need to get fancy with the error return in the future.